contracts/L2/pool/DelegatorPool.sol

```diff
@@ -0,0 +1,103 @@
+//SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+
+import "./MathUtils.sol";
+
+interface IBondingManager {
+    function pendingStake(address _addr, uint256 _endRound)
+        external
+        view
+        returns (uint256);
+
+    function pendingFees(address _addr, uint256 _endRound)
+        external
+        view
+        returns (uint256);
+
+    function transferBond(
+        address _delegator,
+        uint256 _amount,
+        address _oldDelegateNewPosPrev,
+        address _oldDelegateNewPosNext,
+        address _newDelegateNewPosPrev,
+        address _newDelegateNewPosNext
+    ) external;
+
+    function withdrawFees() external;
+}
+
+contract DelegatorPool is Initializable {
+    uint256 public remainingStake;
+    uint256 public cumulativeIncreaseRatio = 1;
+
+    address public bondingManager;
+    address public migrator;
+
+    event Claimed(address _delegator, uint256 _stake, uint256 _fees);
+
+    modifier onlyMigrator() {
+        require(msg.sender == migrator, "DelegatorPool#claim: NOT_MIGRATOR");
+        _;
+    }
+
+    function initialize(address _bondingManager) public initializer {
+        bondingManager = _bondingManager;
+        migrator = msg.sender;
+        remainingStake = pendingStake();
+    }
+
+    receive() external payable {}
+
+    function claim(address _delegator, uint256 _stake) external onlyMigrator {
+        // Calculate original total stake
+        uint256 currTotalStake = pendingStake();
+
+        require(_stake <= currTotalStake, "DelegatorPool#claim: INVALID_STAKE");
+
+        // Calculate Overall Increase ratio
+        cumulativeIncreaseRatio *= currTotalStake / remainingStake;
+
+        // Calculate Stake owed to delegator
+        uint256 owedStake = cumulativeIncreaseRatio * _stake;
```

Comment:
Do we need `cumulativeIncreaseRatio`? Or could a simpler approach that just requires updating a single state var (with the other state var only being set upon initialization):

```
uint256 currTotalStake = pendingStake();
uint256 owedStake = (currTotalStake * stake) / (initTotalStake - initTotalStakeClaimed));

uint256 currTotalFees = pendingFees();
uint256 owedFees = (currTotalFees * stake) / (initTotalStake - initTotalStakeClaimed);

initTotalStakeClaimed += stake;
```

with `initTotalStake` being a state var that is set when this contract is initialized to the delegated stake of the orchestrator that migrated in L2Migrator.finalizeMigrateDelegator(). So, in the above snippet, this contract would just calculate each delegator's % share of `initTotalStake` and the contract can use that % to determine the delegator's share of `currTotalStake` as well as `currTotalFees`. Whenever a delegator claims, `initTotalStakeClaimed` would be incremented so that the % of each remaining delegator increases.